### ISSUE TITLE
formatters.format_ad_timestamp(): Local variable 'timestamp' referenced before assignment 

### ISSUE THREAD

**Issue by klada** [cannatag/ldap3#394]
As you can see [here](https://github.com/cannatag/ldap3/blob/master/ldap3/protocol/formatters/formatters.py#L107) the local variable `timestamp` is used in the exception handler, where it is not defined.

Running in the exception handler raises an `UnboundLocalError`. This happens for example if you want to pass a `datetime` object to the Active Directory.

**Comment by klada** [comment]
This is a semi-minimal example which triggers the bug:

```python
# -*- coding: utf-8 -*-
from __future__ import division, print_function, unicode_literals
import datetime
from ldap3 import Server, Connection, OFFLINE_AD_2012_R2, MOCK_SYNC
from pytz import timezone

DOMAIN = "ad.example.org"
BASE= "dc=ad,dc=example,dc=org"
USERNAME = "username"

tz = timezone("Europe/Berlin")

user_dict = { 
    'objectClass': ['top', 'person', 'organizationalPerson', 'user'],
    'cn': USERNAME,
    'sAMAccountName': USERNAME,
    'userPrincipalName': "{}@{}".format(USERNAME, DOMAIN),
    'accountExpires': datetime.datetime(2018,3,31,23,59,59, tzinfo=tz),
    'profilePath': "\\\\{}\\dfs\\profile\\{}".format(DOMAIN, USERNAME),
    'homeDirectory': "\\\\{}\\dfs\\user\\{}".format(DOMAIN, USERNAME),
    'homeDrive': "H:",
}

server = Server('my_fake_server', get_info=OFFLINE_AD_2012_R2)
connection = Connection(server, '', '', client_strategy=MOCK_SYNC)
# anon fake bind
connection.bind()
dn="cn={},{}".format(USERNAME, BASE)
# Raises error
print(connection.add(dn, [], user_dict))
```
What actually triggers the issue here is the timezone in the `accountExpires`. Actually this is another bug which I was going to report, but I stumbled across this bug here first.

**Comment by cannatag** [comment]
Please check code in dev. It should work now.

**Comment by klada** [comment]
You are right, the code in dev fixes it. Thanks -> you may close this.

**Comment by cannatag** [comment]
Fixed in v2.4
